File: app/src/main/java/net/veldor/flibustaloader/dialogs/GifDialog.java
package net.veldor.flibustaloader.dialogs;

import android.app.Activity;
import android.app.Dialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.TextView;

import net.veldor.flibustaloader.R;

import pl.droidsonroids.gif.GifImageView;

public class GifDialog {

    public static class Builder {
        private String title, message, positiveBtnText;
        private final Activity activity;
        private GifDialogListener pListener;
        private boolean cancel;
        int gifImageResource;

        public Builder(Activity activity) {
            this.activity = activity;
        }

        public Builder setTitle(String title) {
            this.title = title;
            return this;
        }


        public Builder setMessage(String message) {
            this.message = message;
            return this;
        }

        public Builder setPositiveBtnText(String positiveBtnText) {
            this.positiveBtnText = positiveBtnText;
            return this;
        }
/*
        public Builder setPositiveBtnBackground(String pBtnColor) {
            this.pBtnColor = pBtnColor;
            return this;
        }*/

/*
        public Builder setNegativeBtnText(String negativeBtnText) {
            this.negativeBtnText = negativeBtnText;
            return this;
        }*/
/*
        public Builder setNegativeBtnBackground(String nBtnColor) {
            this.nBtnColor = nBtnColor;
            return this;
        }*/

        //set Positive listener
        public Builder OnPositiveClicked(GifDialogListener pListener) {
            this.pListener = pListener;
            return this;
        }
/*
        //set Negative listener
        public Builder OnNegativeClicked(GifDialogListener nListener) {
            this.nListener = nListener;
            return this;
        }*/

        public Builder isCancellable(boolean cancel) {
            this.cancel = cancel;
            return this;
        }

        public Builder setGifResource(int gifImageResource) {
            this.gifImageResource = gifImageResource;
            return this;
        }

        public Dialog build() {
            TextView message1, title1;
            Button nBtn, pBtn;
            GifImageView gifImageView;
            final Dialog dialog = new Dialog(activity);
            dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
            Window window = dialog.getWindow();
            if (window != null)
                window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            dialog.setCancelable(cancel);
            dialog.setContentView(R.layout.gifdialog);


            //getting resources
            title1 = dialog.findViewById(R.id.title);
            message1 = dialog.findViewById(R.id.message);
            nBtn = dialog.findViewById(R.id.negativeBtn);
            pBtn = dialog.findViewById(R.id.positiveBtn);
            gifImageView = dialog.findViewById(R.id.gifImageView);
            gifImageView.setImageResource(gifImageResource);

            title1.setText(title);
            message1.setText(message);
            if (positiveBtnText != null) {
                pBtn.setText(positiveBtnText);

                pBtn.setOnClickListener(view -> {
                    if (pListener != null) pListener.OnClick();
                    dialog.dismiss();
                });
            } else {
                pBtn.setVisibility(View.GONE);
            }
                nBtn.setVisibility(View.GONE);

            return dialog;

        }
    }
}

File: app/src/main/java/net/veldor/flibustaloader/dialogs/GifDialog.kt
package net.veldor.flibustaloader.dialogs

import android.app.Activity
import android.app.Dialog
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.View
import android.view.Window
import android.widget.Button
import android.widget.TextView
import net.veldor.flibustaloader.R
import pl.droidsonroids.gif.GifImageView

class GifDialog {
    class Builder(private val activity: Activity) {
        private var title: String? = null
        private var message: String? = null
        private var positiveBtnText: String? = null
        private var pListener: GifDialogListener? = null
        private var cancel = false
        private var gifImageResource = 0
        fun setTitle(title: String?): Builder {
            this.title = title
            return this
        }

        fun setMessage(message: String?): Builder {
            this.message = message
            return this
        }

        fun setPositiveBtnText(positiveBtnText: String?): Builder {
            this.positiveBtnText = positiveBtnText
            return this
        }

        /*
        public Builder setPositiveBtnBackground(String pBtnColor) {
            this.pBtnColor = pBtnColor;
            return this;
        }*/
        /*
        public Builder setNegativeBtnText(String negativeBtnText) {
            this.negativeBtnText = negativeBtnText;
            return this;
        }*/
        /*
        public Builder setNegativeBtnBackground(String nBtnColor) {
            this.nBtnColor = nBtnColor;
            return this;
        }*/
        //set Positive listener
        fun onPositiveClicked(pListener: GifDialogListener?): Builder {
            this.pListener = pListener
            return this
        }

        /*
        //set Negative listener
        public Builder OnNegativeClicked(GifDialogListener nListener) {
            this.nListener = nListener;
            return this;
        }*/
        fun isCancellable(cancel: Boolean): Builder {
            this.cancel = cancel
            return this
        }

        fun setGifResource(gifImageResource: Int): Builder {
            this.gifImageResource = gifImageResource
            return this
        }

        fun build(): Dialog {
            val message1: TextView
            val title1: TextView
            val nBtn: Button
            val pBtn: Button
            val gifImageView: GifImageView
            val dialog = Dialog(activity)
            dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
            val window = dialog.window
            window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            dialog.setCancelable(cancel)
            dialog.setContentView(R.layout.gifdialog)


            //getting resources
            title1 = dialog.findViewById(R.id.title)
            message1 = dialog.findViewById(R.id.message)
            nBtn = dialog.findViewById(R.id.negativeBtn)
            pBtn = dialog.findViewById(R.id.positiveBtn)
            gifImageView = dialog.findViewById(R.id.gifImageView)
            gifImageView.setImageResource(gifImageResource)
            title1.text = title
            message1.text = message
            if (positiveBtnText != null) {
                pBtn.text = positiveBtnText
                pBtn.setOnClickListener {
                    if (pListener != null) pListener!!.onClick()
                    dialog.dismiss()
                }
            } else {
                pBtn.visibility = View.GONE
            }
            nBtn.visibility = View.GONE
            return dialog
        }
    }
}